OpenRA.Mods.Common/Traits/Tooltip.cs

```diff
@@ -32,10 +32,17 @@ public class TooltipInfo : ITraitInfo, ITooltipInfo
 		public readonly bool ShowOwnerRow = true;
 
 		[Desc("Sequence of the actor that contains the cameo.")]
-		public readonly string Icon = "icon";
+		public string Icon { get; private set; }
```

Comment:
An alternative solution may be to make `Tooltip` upgradable, and then have the build palette code pick the one that is allowed based on the current upgrades (you'll need new code for that however - the current upgrade checking code only exists for living actors).